apps/wallet/src/ui/app/components/accounts/UnlockAccountContext.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { createContext, useContext, useState, type ReactNode, useCallback } from 'react';
+
+import { toast } from 'react-hot-toast';
+import { UnlockAccountModal } from './UnlockAccountModal';
+import { useAccounts } from '../../hooks/useAccounts';
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { useUnlockMutation } from '../../hooks/useUnlockMutation';
+
+interface UnlockAccountContextType {
+	isUnlockModalOpen: boolean;
+	accountIdToUnlock: string | null;
+	unlockAccount: (accountId: string) => void;
+	isLoading: boolean;
+	lockAccount: (accountId: string) => void;
+	hideUnlockModal: () => void;
+}
+
+const UnlockAccountContext = createContext<UnlockAccountContextType | undefined>(undefined);
+
+export const UnlockAccountProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+	const [isUnlockModalOpen, setIsUnlockModalOpen] = useState(false);
+	const [accountIdToUnlock, setAccountIdToUnlock] = useState<string | null>(null);
+	const { data: accounts } = useAccounts();
+	const unlockAccountMutation = useUnlockMutation();
+	const backgroundClient = useBackgroundClient();
+	const hideUnlockModal = useCallback(() => {
+		setIsUnlockModalOpen(false);
+		setAccountIdToUnlock(null);
+	}, []);
+
+	const unlockAccount = useCallback(
+		async (accountId: string) => {
+			const account = accounts?.find((account) => account.id === accountId);
+			if (account) {
+				if (account.isPasswordUnlockable) {
+					// account unlocking will happen in modal after password is entered
+					setIsUnlockModalOpen(true);
+					setAccountIdToUnlock(accountId);
+				} else {
+					try {
+						// try unlocking non-password unlockable accounts immediately
+						await unlockAccountMutation.mutate({ id: accountId });
```

Comment:
nit: I'd do the success and error handling in the mutation callbacks instead of this try catch block (I also thought `mutate` wasn't async). 

https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync

```
unlockAccountMutation.mutate({ id: accountId }, {
  onSuccess: () => toast.success('Account unlocked'),
  onError: (e) => toast.error((e as Error).message),
});
```

Reply:
yeah this should change to mutateAsync